privmx-endpoint/src/main/cpp/modules/StoreApi.cpp

```diff
@@ -613,58 +613,121 @@ Java_com_simplito_java_privmx_1endpoint_modules_store_StoreApi_closeFile(
+                    subscription_queries_c.push_back(ctx.jString2string((jstring) arrayElement));
+                }
+
+                auto subscription_ids_c = getStoreApi(ctx, thiz)->subscribeFor(
+                        subscription_queries_c);
+
+                jobject array = env->NewObject(arrayListCls, initMID);
+                for (auto &id: subscription_ids_c) {
+                    ctx->CallObjectMethod(
+                            array,
+                            addToListMID,
+                            ctx->NewStringUTF(id.c_str())
+                    );
+                }
+                return array;
+            }
+    );
+    if (ctx->ExceptionCheck()) {
+        return nullptr;
+    }
+    return result;
 }
+
 extern "C"
 JNIEXPORT void JNICALL
-Java_com_simplito_java_privmx_1endpoint_modules_store_StoreApi_unsubscribeFromStoreEvents(
+Java_com_simplito_java_privmx_1endpoint_modules_store_StoreApi_unsubscribeFrom(
         JNIEnv *env,
-        jobject thiz
+        jobject thiz,
+        jobject subscription_ids
 ) {
     JniContextUtils ctx(env);
-    ctx.callVoidEndpointApi([&ctx, &thiz]() {
-        getStoreApi(ctx, thiz)->unsubscribeFromStoreEvents();
+    if (ctx.nullCheck(subscription_ids, "Subscription IDs")) {
+        return;
+    }
+
+    ctx.callVoidEndpointApi([&ctx, &thiz, &subscription_ids]() {
+        auto subscription_ids_arr = ctx.jObject2jArray(subscription_ids);
+        auto subscription_ids_c = std::vector<std::string>();
+
+        for (int i = 0; i < ctx->GetArrayLength(subscription_ids_arr); i++) {
+            jobject arrayElement = ctx->GetObjectArrayElement(subscription_ids_arr, i);
+            subscription_ids_c.push_back(ctx.jString2string((jstring) arrayElement));
+        }
+
+        getStoreApi(ctx, thiz)->unsubscribeFrom(subscription_ids_c);
     });
 }
+
 extern "C"
-JNIEXPORT void JNICALL
-Java_com_simplito_java_privmx_1endpoint_modules_store_StoreApi_subscribeForFileEvents(
+JNIEXPORT jstring JNICALL
+Java_com_simplito_java_privmx_1endpoint_modules_store_StoreApi_buildSubscriptionQuery(
         JNIEnv *env,
         jobject thiz,
-        jstring store_id
+        jlong eventType,
+        jlong selectorType,
+        jstring selectorId
 ) {
     JniContextUtils ctx(env);
-    if (ctx.nullCheck(store_id, "Store ID")) {
-        return;
+    if (ctx.nullCheck(selectorId, "SelectorID")) {
+        return nullptr;
     }
-    ctx.callVoidEndpointApi([&ctx, &thiz, &store_id]() {
-        getStoreApi(ctx, thiz)->subscribeForFileEvents(
-                ctx.jString2string(store_id)
-        );
-    });
+
+    jstring result;
+    ctx.callResultEndpointApi<jstring>(
+            &result,
+            [&ctx, &thiz, &eventType, &selectorType, &selectorId]() {
+
+                auto result = getStoreApi(ctx, thiz)->buildSubscriptionQuery(
+                        parseStoreEventType(ctx, (long) eventType),
+                        parseStoreEventSelectorType(ctx, (long) selectorType),
```

Comment:
you can easy cast long value to enum like this
```suggestion
                        static_cast<store::EventSelectorType>(selectorType);
```